det3d/datasets/argoverse/argoverse_common.py

```diff
@@ -0,0 +1,330 @@
+
+    existing_boxes = torch.tensor([track[t].boxes for t in timestamps], dtype=torch.float32)
+    existing_boxes_tensor = convert_track_to_tensor(existing_boxes, track)
+
+    if len(to_interpolate) == 0:
+        # Nothing to interpolate
+        return existing_boxes_tensor, []
+    elif len(timestamps) == 1:
+        # Not enough to interpolate. Repeat the tensor n_t times
+        return existing_boxes_tensor.repeat((n_t, 1)), to_interpolate
+
+    # Fit rotations
+    rots = Rotation.from_euler('x', existing_boxes_tensor[:, 6])
+    spline = RotationSpline(timestamps, rots)
+    interpolated_rots = spline(to_interpolate)
+
+    # Fit x, y and z of translation separately
+    # UnivariateSpline works the best practically in our case, but it doesn't work with <= 3 points
+    interp_cls = interpolate.UnivariateSpline if len(timestamps) > 3 else interpolate.KroghInterpolator
+    func_x = interp_cls(timestamps, existing_boxes_tensor[:, 0])
+    func_y = interp_cls(timestamps, existing_boxes_tensor[:, 1])
+    func_z = interp_cls(timestamps, existing_boxes_tensor[:, 2])
+
+    interpolated_x = func_x(to_interpolate)
+    interpolated_y = func_y(to_interpolate)
+    interpolated_z = func_z(to_interpolate)
+
+    interpolated = torch.zeros((n_t, 7))  # 7 values per bounding box
+
+    interpolated[timestamps, :] = existing_boxes_tensor
+
+    first_label = track[list(iter(track))[0]]
+
+    interpolated[to_interpolate, 0] = torch.tensor(interpolated_x, dtype=torch.float32)
+    interpolated[to_interpolate, 1] = torch.tensor(interpolated_y, dtype=torch.float32)
+    interpolated[to_interpolate, 2] = torch.tensor(interpolated_z, dtype=torch.float32)
+
+    # Length, width, height and label are set to the same value as the first given label
+    interpolated[to_interpolate, 3] = first_label.record.length
+    interpolated[to_interpolate, 4] = first_label.record.width
+    interpolated[to_interpolate, 5] = first_label.record.height
+
+    interpolated[to_interpolate, 6] = torch.tensor(interpolated_rots.as_euler('xyz')[:, 0], dtype=torch.float32)
+
+    return interpolated, to_interpolate
+
+def convert_track_to_tensor(bboxes, track):
+    center_and_yaws = find_center_and_rotation(bboxes)
+    # TODO: centerpoint expects wlh?
+    l_w_h = torch.tensor([
+        [label.record.length, label.record.width, label.record.height]
+        for _, label in track.items()
+    ], dtype=torch.float32)
+    return torch.cat((
+        center_and_yaws[:, 0].unsqueeze(1),
+        center_and_yaws[:, 1].unsqueeze(1),
+        bboxes[:, 0, 2].unsqueeze(1),
+        l_w_h,
+        center_and_yaws[:, 2].unsqueeze(1),
+    ), dim=1)
+
+def find_center_and_rotation(boxes: torch.Tensor) -> torch.Tensor:
+    """
+    Finds the (x,y) coordinates of the bounding boxes' centers, as well as their yaws
+
+    Args:
+        boxes: (torch.Tensor) A list of bounding boxes, each bounding box has four corners (x, y, z)
```

Comment:
could you add a description of the dimensions of the boxes tensor, and which tensor corresponds to?